DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -55,13 +57,24 @@ public IActionResult Index()
                 return View();
             }
 
+            var email = delegateRegistrationData.RegisterViewModel.Email;
+            if (email != null && userService.GetUsersByEmailAddress(email).delegateUsers.Count != 0)
+            {
+                ModelState.AddModelError(nameof(RegisterViewModel.Email), "A user with this email address already exists");
+            }
+
             return View(delegateRegistrationData.RegisterViewModel);
         }
 
         [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationData>))]
         [HttpPost]
         public IActionResult Index(RegisterViewModel model)
         {
+            if (model.Email != null && userService.GetUsersByEmailAddress(model.Email).delegateUsers.Count != 0)
+            {
+                ModelState.AddModelError(nameof(RegisterViewModel.Email), "A user with this email address already exists");
+            }

```

Comment:
Could we pull this into a private method ValidateEmailAddress or similar that can be called both here and above?